src/Param.cc

```diff
@@ -65,54 +65,66 @@ Param::Param(const std::string &_key, const std::string &_typeName,
              const std::string &_description)
   : dataPtr(new ParamPrivate)
 {
-  this->dataPtr->key = _key;
-  this->dataPtr->required = _required;
-  this->dataPtr->typeName = _typeName;
-  this->dataPtr->description = _description;
-  this->dataPtr->set = false;
-  this->dataPtr->ignoreParentAttributes = false;
-  this->dataPtr->defaultStrValue = _default;
+  sdf::Errors errors;
+  this->dataPtr->Init(_key, _typeName, _default, _required,
+                      errors, _description);
 
-  SDF_ASSERT(
-      this->dataPtr->ValueFromStringImpl(
-          this->dataPtr->typeName,
-          _default,
-          this->dataPtr->defaultValue),
-      "Invalid parameter");
-  this->dataPtr->value = this->dataPtr->defaultValue;
-  this->dataPtr->strValue = std::nullopt;
+  if(!errors.empty())
+  {
+    for (unsigned int i = 0; i < errors.size() - 1; ++i)
+    {
+      sdferr << errors[i].Message() + "\n";
+    }
+
+    SDF_ASSERT(false, errors.back().Message());
```

Comment:
what is the purpose of this assert?

Reply:
The idea was to keep the same behavior as before (see https://github.com/ignitionrobotics/sdformat/blob/e363110e9c5a3671acb371537a6eaef5617f1d28/src/Param.cc#L106 ). Basically, f there are any errors in the vector use the assert to throw the exception with the last added message as it was done previously.

Reply:
thanks, I hadn't followed the changes closely enough, but I get it now!